File: baekjoon/BinarySearch/13706_제곱근.py
import sys
input = sys.stdin.readline
N = int(input())


def solution():
    answer = 0
    start = 1
    end = N
    while True:
        mid = (start+end) // 2
        if mid * mid == N:
            answer = mid
            break
        elif mid * mid > N:
            end = mid-1
        elif mid * mid < N:
            start = mid+1
    return answer


print(solution())

File: baekjoon/BinarySearch/13706_제곱근.js
const fs = require('fs')
const N = Number(fs.readFileSync('../input.txt').toString().trim())

function solution() {
  let answer = 0
  let start = 1
  let end = N

  while (true) {
    let mid = parseInt((start + end) / 2)

    if (mid * mid === N) {
      answer = mid
      break
    } else if (mid * mid > N) {
      end = mid - 1
    } else if (mid * mid < N) {
      start = mid + 1
    }
  }

  return answer
}

console.log(solution())
